apps/codelab/src/app/directives/access/access.directive.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+@Directive({
+  selector: '[access]'
```

Comment:
As I mentioned before, rename it and inject the service here, this would improve the UX a lot